src/charm.py

```diff
@@ -28,7 +28,7 @@
 # Juju defines the value of this label.
 # It has the same value as the label "app.kubernetes.io/name"
 # set in the service account associated with the application.
-CREATED_BY_LABEL = "app.juju.is/created-by="
+CREATED_BY_LABEL = "Ingress-label"
```

Comment:
Also, how would this work on upgrades? E.g., when an application currently using the `app.juju.is/created-by=` label upgrades to this version of the charm, will everything still work? Should we have an integration test to confirm the upgrade?

Reply:
The upgrade process is not yet done, that's why this PR is still a draft. I need the PR to test in the CI since it throws different results locally. Also will correct the rest of your comments after the tests start passing.